system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -41,7 +41,10 @@ function countTopSitesIconsTypes(topSites) {
 }
 
 class TopSites extends React.PureComponent {
-  componentDidUpdate() {
+  /**
+   * Dispatch session statistics about the quality of TopSites icons.
+   */
+  _storeTopSitesIconStats() {
     const realTopSites = this.props.TopSites.rows.slice(0, this.props.TopSitesCount);
```

Comment:
What does "real" mean here? Are the ones sliced away not real? If it's supposed to be the same functionality as that of `render`, then let's refactor that.

Reply:
The way I would do this is set `this.realTopSites` in the `render` method. Alternatively there is `componentWillUpdate` but docs say:
> This method is not called for the initial render.

Reply:
Saving the property is almost like setting state, which shouldn't be done from `render`. And I believe the react pattern is to not set state based on props and instead just provide a helper to compute the value from props.

Reply:
Also, still doesn't answer what "real" means here.

Reply:
That was the name used in the TopSites render method. We have two sets of top sites: the ones you regularly see and the ones in the top sites edit modal (that’s where the distinction came from I think). @r1cky knows more about this.